contrib/net/http/utils.go

```diff
@@ -0,0 +1,73 @@
+package http
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/DataDog/dd-trace-go/tracer/ext"
+)
+
+type ServeHTTP func(http.ResponseWriter, *http.Request)
+
+// Trace will trace the request before calling the ServeHTTP function passed in argument.
+func Trace(serveHTTP ServeHTTP, w http.ResponseWriter, r *http.Request, service, resource string, t *tracer.Tracer) {
```

Comment:
can't you replace `ServeHTTP` with `http.Handler` ?

Reply:
Yeah you could effectively pass a `http.Handler` instead of the `ServeHTTP` custom type.
Something like below would work fine:
```go
func Trace(h http.Handler, w http.ResponseWriter, r *http.Request, service, resource string, t *tracer.Tracer) {
	// bail out if tracing isn't enabled
	if !t.Enabled() {
		serveHTTP(w, r)
		return
	}

	// TODO: get the span from the request context
	span := t.NewRootSpan("http.request", service, resource)
	defer span.Finish()

	span.Type = ext.HTTPType
	span.SetMeta(ext.HTTPMethod, r.Method)
	span.SetMeta(ext.HTTPURL, r.URL.Path)

	// pass the span through the request context
	ctx := span.Context(r.Context())
	traceRequest := r.WithContext(ctx)

	// trace the response to get the status code
	traceWriter := NewResponseWriter(w, span)

	// serve the request to the underlying multiplexer
	h.ServeHTTP(traceWriter, traceRequest)
}
```

Reply:
I think it's better to use `http.Handler` everywhere.
The type `ServeHTTP`/`http.HandlerFunc` is too restrictive.

Reply:
Done